tests/gnrc_netif_ipv6_wait_for_global_address/main.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Tests notification for global address
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+
+#include "embUnit.h"
+#include "embUnit/embUnit.h"
+#include "net/ipv6.h"
+#include "net/gnrc/netif.h"
+#include "net/gnrc/netif/raw.h"
+#include "net/netdev_test.h"
+#include "xtimer.h"
+
+#define TEST_NETIF_NUMOF    2
+#define TEST_NETIF_PRIO     3
+
+static netdev_test_t netdev_test[TEST_NETIF_NUMOF];
+static gnrc_netif_t netif_test[TEST_NETIF_NUMOF];
+static char netif_stack[TEST_NETIF_NUMOF][THREAD_STACKSIZE_DEFAULT];
+static char adder_stack[THREAD_STACKSIZE_DEFAULT];
+
+static gnrc_netif_t *_test_netif;
+static const ipv6_addr_t _test_addr =  {{ 0x20, 0x01, 0x0d, 0xd8,
+                                          0x00, 0x00, 0x00, 0x00,
+                                          0x00, 0x00, 0x00, 0x00,
+                                          0x00, 0x00, 0x00, 0x01 }};
+static void tear_down(void)
+{
+    if (_test_netif == NULL) {
+        return;
+    }
+
+    gnrc_netif_ipv6_addr_remove(_test_netif, &_test_addr);
+    _test_netif = NULL;
+}
+
+static void *_adder_thread(void *netif)
+{
+    xtimer_msleep(10);
+    gnrc_netif_ipv6_addr_add(netif, &_test_addr, 64,
+                             GNRC_NETIF_IPV6_ADDRS_FLAGS_STATE_VALID);
+    return NULL;
+}
+
+static void _add_delayed_addr(gnrc_netif_t *netif)
+{
+    _test_netif = netif;
+    memset(adder_stack, 0, sizeof(adder_stack));
+    thread_create(adder_stack, sizeof(adder_stack),
+                  THREAD_PRIORITY_MAIN - 1,
+                  THREAD_CREATE_STACKTEST,
+                  _adder_thread, netif, "add_addr");
+}
+
+static void test_wait_timeout(void)
+{
+    TEST_ASSERT_EQUAL_INT(TEST_NETIF_NUMOF, gnrc_netif_numof());
+    TEST_ASSERT(!gnrc_netif_ipv6_wait_for_global_address(NULL, 10));
+    TEST_ASSERT(!gnrc_netif_ipv6_wait_for_global_address(&netif_test[0], 10));
+}
+
+static void test_wait_timeout_other_iface(void)
+{
+    unsigned now;
+
+    TEST_ASSERT_EQUAL_INT(TEST_NETIF_NUMOF, gnrc_netif_numof());
+
+    /* no event when adding addr to other interface */
+    _add_delayed_addr(&netif_test[1]);
+    now = xtimer_now_usec();
+    TEST_ASSERT(!gnrc_netif_ipv6_wait_for_global_address(&netif_test[0], 20));
+    TEST_ASSERT(((xtimer_now_usec() - now) / 1000) >= 20);
+}
+
+static void test_wait_success(void)
+{
+    unsigned now;
+
+    /* event when adding addr to specified interface */
+    now = xtimer_now_usec();
+    _add_delayed_addr(&netif_test[0]);
+    TEST_ASSERT(gnrc_netif_ipv6_wait_for_global_address(&netif_test[0], 20));
```

Comment:
Last thing (hopefully): This can be its own function now and called at the end of the `success` test cases:

```C
static void _assert_wait_blocks(gnrc_netif_t *add_netif,
                                gnrc_netif_t *wait_netif,
                                uint32_t timeout)
{
    uint32_t now = xtimer_now_usec();

    _add_delayed_addr(add_netif);
    TEST_ASSERT(gnrc_netif_ipv6_wait_for_global_address(wait_netif,
                                                        timeout));
    TEST_ASSERT(((xtimer_now_usec() - now) / US_PER_MS) < timeout);
}
```

(Sorry I did not include the last line into my commenting... Github did not let me comment on that line range for some reason...)

Reply:
Ok